apps/backend/src/app/modules/maps/discord-webhook.service.ts

```diff
@@ -0,0 +1,181 @@
+      }
+    });
+
+    const mainTrackLeaderboards = extendedMap.leaderboards.filter(
+      (lb) =>
+        lb.trackType === TrackType.MAIN && lb.type !== LeaderboardType.HIDDEN
+    );
+
+    const mainTrackGamemodes = new Set(
+      mainTrackLeaderboards.map((sugg) => sugg.gamemode)
+    );
+
+    const mapAuthors = extendedMap.credits
+      .filter((cred) => cred.type === MapCreditType.AUTHOR)
+      .map((s) => s.user.alias)
+      .sort();
+
+    const webhookBody = this.createWebhookBody(
+      'New map has been approved!',
+      extendedMap,
+      mapAuthors,
+      mainTrackLeaderboards
+    );
+
+    await this.broadcastWebhookToCategories(mainTrackGamemodes, webhookBody);
+  }
+
+  createWebhookBody(
+    text: string,
+    extendedMap: MMap & { info: MapInfo; submitter: User },
+    authors: Array<string>,
+    gamemodes: MapSubmissionSuggestion[] | Leaderboard[]
+  ) {
+    const FRONTEND_URL = this.config.getOrThrow('url.frontend');
+    const CDN_URL = this.config.getOrThrow('url.cdn');
+
+    return {
+      content: text,
+      embeds: [
+        {
+          title: extendedMap.name,
+          description: 'By ' + authors.map((a) => `**${a}**`).join(', '),
+          url: `${FRONTEND_URL}/maps/${extendedMap.id}`,
+          timestamp: extendedMap.info.creationDate.toISOString(),
+          color: 1611475,
+          image: {
+            url: `${CDN_URL}/${imgLargePath(extendedMap.images[0])}`
+          },
+          fields: gamemodes.map(
+            (gm: MapSubmissionSuggestion | Leaderboard) => ({
+              name: GamemodeInfo.get(gm.gamemode).name,
+              value: `Tier ${gm.tier}${gm.type === LeaderboardType.UNRANKED ? ' (Unranked)' : ''}`,
+              inline: true
+            })
+          ),
+          footer: {
+            icon_url: `https://avatars.cloudflare.steamstatic.com/${extendedMap.submitter.avatar}.jpg`,
+            text: extendedMap.submitter.alias
+          }
+        }
+      ]
+    };
+  }
+
+  async broadcastWebhookToCategories(
+    gamemodes: Set<Gamemode>,
+    webhookBody: object
+  ) {
+    for (const [category, modes] of GamemodeCategories.entries()) {
+      if (modes.some((gm) => gamemodes.has(gm))) {
+        const webhookUrl = this.config.getOrThrow(
+          `discordWebhooks.${category}`
+        );
+
+        if (webhookUrl === '') continue;
+
+        const responseOrStatus = await this.postWebhook(
+          webhookUrl,
+          webhookBody
+        );
+
+        if (!responseOrStatus) break;
+      }
+    }
+  }
+
+  async postWebhook(webhookUrl: string, body: object) {
+    return await firstValueFrom(
+      this.http.post(webhookUrl, body, {
+        headers: { 'Content-Type': 'application/json' }
+      })
+    ).catch((error) => {
+      this.logger.error(
+        'Failed to post to webhook',
+        error.response ? JSON.stringify(error.response.data, null, 2) : error
+      );
+      return false;
+    });
+  }
+}
```

Comment:
`postWebhook` is doing very little here, and I don't see the point of catching this error to return false, so you can handle `responseOrStatus` using a boolean check rather than an exception. If you avoid handling the exception in the inner section, you can just use `Promise.all`'s behaviour to get the same behaviour if a request errors (stop all requests)

```ts
  async broadcastWebhookToCategories(
    gamemodes: Set<Gamemode>,
    webhookBody: object
  ): Promise<void> {
    await Promise.all(
      [...GamemodeCategories.entries()]
        .filter(([, modes]) => modes.some((gm) => gamemodes.has(gm)))
        .map(([category]) => {
          const webhookUrl = this.config.getOrThrow(
            `discordWebhooks.${category}`
          );

          if (webhookUrl === '') return;

          return this.http.post(webhookUrl, webhookBody, {
            headers: { 'Content-Type': 'application/json' }
          });
        })
    ).catch((error) => this.logger.error('Failed to post to webhook', error));
  }
```